### ISSUE TITLE
Question: How to sort and delete multiple files

### ISSUE THREAD

**Issue by Brandon-Campanile** [andreafabrizi/Dropbox-Uploader#457]
I'm not very familiar with bash, but I was wondering if the following lines would work with Dropbox Uploader:
`$oldFiles = $DBuploader list '/.PCbackup' | awk '{print $2}' | sort -nr | awk 'NR>2'
        $DBuploader delete '/.PCbackup/$oldFiles'`
I have multiple files in my dropbox's /.PCbackup directory named 'backup-$somedate'. The goal is to get the list of files, sort them from new->old, and delete all but the newest two.

Please let me know if something like this would work or if i'd have to do a while-do loop. Thanks

**Comment by testaccountsnt** [comment]
The better way of doing it, write another bash script which will provide desired result. You can put date tag in file name what you are uploading on dropbox, if there is name in the file, it will be easier to remove unwanted files.

Here take a look at one example.
https://stackoverflow.com/questions/41098278/delete-files-after-certain-date-with-dropbox-uploader-sh

**Comment by Brandon-Campanile** [comment]
@testaccountsnt I had seen that example. I already have the date in the filename "Backup-($date+%Y-%m-%d)", but the solution you linked is pretty inelegant. I mean, converting the date to seconds? I was hoping for a one line command. Will the above not work? I just need to know if dropbox uploader can handle multiple files like this.

**Comment by testaccountsnt** [comment]
If you do not upload same filename again and each time file name is different, it lists files by modified date default.
If you sort it by -n -r it will sort by alphabetical order recursively. You can use for loop to delete those files.

**Comment by Brandon-Campanile** [comment]
That's what I was wondering. So the above will work if within a loop. Thank you.

**Comment by testaccountsnt** [comment]
Yes, I am suggesting loop method because it is not deleting multiple files using DELETE option just like it uploads multiple files using UPLOAD option. Still it can be done in two steps on terminal without writing any script for that.

## awk print 3 will print the third column which is filenames on my system. tail -n +4 starts printing the result from fourth row, it skips first row that is "/" and next two rows which are files which you do not want to delete. - Command 1
list=$(/home/ubuntu/Dropbox-Uploader/dropbox_uploader.sh list | awk '{print $3}' | tail -n +4)

### IFS is internal field separator which handles the condition if there are spaces in filename. Now delete all files from Row 4 - Command 2
IFS=$'\n' && for i in $list; do /home/ubuntu/Dropbox-Uploader/dropbox_uploader.sh delete $i; done

I hope it will help.

**Comment by Brandon-Campanile** [comment]
It does help, thank you. I'm trying to write a backup script with encrytion. I was also wondering if it can upload from a stream like so:
```
Dropbox_uploader.sh mkdir /dropbox/path/backup-($date +%Y-%m-%d)/
dropbox_uploader.sh <( openssl enc -e -aes-256-cbc -salt -pass file:'/passfile.txt' <( tar -cpf - --one-file-system -C '/' | split -b 4G - )) /dropbox/path/backup-($date +%Y-%m-%d)/
